src/components/config/index.ts

```diff
@@ -85,6 +74,103 @@ const handshakeAndValidate = async (config: Config) => {
   }
 }
 
+const updateConfig = async (config: Config) => {
+  await handshakeAndValidate(config)
+  const lastConfig = cfg?.version
+  cfg = config
+  cfg.version = lastConfig || md5Hash(config)
+  if (lastConfig !== cfg.version) {
+    emitter.emit(events.config.updated, cfg)
+    log.warn('config file update detected')
+  }
+}
+
+const mergeConfigs = (): Config => {
+  const mergedConfig = configs.reduce((prev, current) => {
+    return {
+      certificate: current?.certificate
+        ? current.certificate
+        : prev.certificate,
+      interval: current?.interval ? current.interval : prev.interval,
+      notifications:
+        current?.notifications && current.notifications.length > 0
+          ? current.notifications
+          : prev.notifications,
+      probes:
+        current?.probes && current.probes.length > 0
+          ? current.probes
+          : prev.probes,
+      symon: current?.symon ? current.symon : prev.symon,
+      'status-notification': current?.['status-notification']
+        ? current['status-notification']
+        : prev['status-notification'],
+    }
+  })
```

Comment:
Since the next config will override the previous one, can we not just use

```
configs.reduce((prev, curr) => {
   return {
      ...prev,
      ...curr
   }
})
```